### ISSUE TITLE
Space names can be covered by a large (10K+) notification badge

### ISSUE THREAD

**Issue by anoadragon453** [vector-im/element-web#18694]
### Steps to reproduce

Create a Space with a relatively long name, and have at least 10.0K notifications in it.

### What happened?

I have a Space containing Subspaces called "Discord Servers". These often contain quite a lot of notifications. The UI looks good when the spaces panel is expanded to show subspaces:

![image](https://user-images.githubusercontent.com/1342360/130457674-ed074d42-d8f3-4be9-9a27-bd71d7b00e96.png)

yet when the panel is expanded and with Subspaces hidden, the notification count covers the space name in a non-aesthetically pleasing way:

![image](https://user-images.githubusercontent.com/1342360/130457850-c24130a4-efb2-4ec5-a27d-03f7f102e4ed.png)

### What did you expect?

It looks like the panel will actually expand depending on the size of the name, and then eventually be cut off with an ellipsis at a certain threshold. For instance, if I change the name of the space to "Discord Servers That I'm In", I see:

![image](https://user-images.githubusercontent.com/1342360/130458347-daade803-8692-424e-b552-2621424c113d.png)

Which sort of works, though I'd expect the be able to see the third dot in the ellipsis there.

TL;DR it looks like the large width of the 3 character notification dot was not expected when designing the margins.

### Operating system

Arch Linux

### Application version

Element Desktop 1.7.34

### How did you install the app?

The AUR

### Have you submitted a rageshake?

No

**Comment by t3chguy** [comment]
Well in this instance, there's not much reason for this to be absolutely positioned, we can rely on the flexbox to lay this out properly instead.
